nifi-toolkit/nifi-toolkit-client/src/main/java/org/apache/nifi/toolkit/client/impl/JerseyProcessGroupClient.java

```diff
@@ -378,4 +381,35 @@ public PasteResponseEntity paste(String processGroupId, PasteRequestEntity paste
                     PasteResponseEntity.class);
         });
     }
+
+    @Override
+    public ProcessGroupEntity upload(String parentPgId, File file, String pgName, Double posX, Double posY) throws NiFiClientException, IOException {
+        if (StringUtils.isBlank(parentPgId)) {
+            throw new IllegalArgumentException("Parent process group id cannot be null or blank");
+        }
+        if (file == null) {
+            throw new IllegalArgumentException("File cannot be null");
+        }
+        if (!file.exists() || !file.canRead()) {
+            throw new IllegalArgumentException("File does not exist: " + file.getAbsolutePath());
```

Comment:
This message is misleading if the issue is `!file.canRead()` as the file may exist yet there are no read permissions.